contracts/dao/neutron-chain-manager/src/msg.rs

```diff
@@ -0,0 +1,137 @@
+use cosmwasm_schema::{cw_serde, QueryResponses};
+use cosmwasm_std::{Addr, CosmosMsg};
+use neutron_sdk::bindings::msg::{NeutronMsg, ParamChange};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, JsonSchema, Debug, Clone)]
+pub struct InstantiateMsg {
+    /// Defines the initial strategy. Must be an ALLOW_ALL strategy.
+    pub initial_strategy: Strategy,
+}
+
+#[derive(Serialize, Deserialize, JsonSchema, Debug, Clone)]
+#[serde(rename_all = "snake_case")]
+pub enum ExecuteMsg {
+    AddStrategy {
+        strategy: Strategy,
+    },
+    RemoveStrategy {
+        address: Addr,
+    },
+    ExecuteMessages {
+        messages: Vec<CosmosMsg<NeutronMsg>>,
+    },
+}
+
+#[cw_serde]
+#[derive(QueryResponses)]
+pub enum QueryMsg {
+    #[returns(Vec < Strategy >)]
+    Strategies {},
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+pub struct MigrateMsg {}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+pub struct Strategy {
+    pub address: Addr,
+    pub permissions: Vec<Permission>,
```

Comment:
Actually it can work with a couple of assumtions, and does not look very weird.
1) Only one entity of each type of permission into strategy is allowed
2) you no need standalone AllowAll enum
```rust
pub struct Strategy {
   permissions: HashMap<Permission, Permission>;
}


impl Hash for Permission {
    fn hash<H: Hasher>(&self, state: &mut H) {
        match self {
            Permission::AllowAll => state.write("AllowAll".as_bytes()),
            Permission::ParamChangePermission(_) => state.write("ParamChangePermission".as_bytes()),
            Permission::UpdateParamsPermission(_) => {}
            Permission::CronPermission(_) => {}
        }
    }
}
```
3) you can even try to delegate validating the messages over the permission to permissions themself.
```rust
impl Strgategy {
   fn find_permission(&self,msg: CosmosMsg);
   fn validate(&self,msg: CosmosMsg) {
       let perm = self.find_permission(msg)
       perm.validate(msg)
   }
}